src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -1,116 +1,62 @@
 import React, {
-  createRef,
   forwardRef,
   useEffect,
   useImperativeHandle,
+  useRef,
   useState,
 } from 'react';
 import PropTypes from 'prop-types';
 import useFetch from '@bloodyaugust/use-fetch';
-import ListOption from '../list-option/list-option.jsx';
-import Loader from '../loader/loader.jsx';
-import Select from '../select/select.jsx';
-import mockConstants from '../../mocks/mock-constants.js';
+import Autocompleter from '../autocompleter/autocompleter.jsx';
 import useForwardedRef from '../../hooks/use-forwarded-ref.js';
+import { API_ROOT } from '../../mocks/mock-constants.js';
 
-const { API_ROOT } = mockConstants;
-const defaultValue = ['-1'];
-
-const CustomFieldInputSingleChoice = forwardRef(function CustomFieldInputSingleChoice(props, ref) {
-  const [choices, setChoices] = useState([]);
-  const listOptionRefs = choices.map(() => createRef());
-  const [value, setValue] = useState(props.value);
-  const selfRef = useForwardedRef(ref);
+const CustomFieldInputSingleChoice = forwardRef(function CustomFieldInputSingleChoice(props, forwardedRef) {
+  const ref = useForwardedRef(forwardedRef);
+  const autocompleterRef = useRef();
   const { execute } = useFetch();
+  const [value, setValue] = useState(undefined);
 
-  function selectOnChangeHandler(event) {
-    if (event.target.value) {
-      setValue([event.target.value.id]);
+  useEffect(() => {
+    if (props.value.length) {
+      execute(`${API_ROOT}/custom_field_choices?for_custom_fields=${props.customFieldID}&only=${props.value[0]}`)
+        .then(({ json, mounted }) => {
+          if (!mounted) return;
+          setValue(json.custom_field_choices[json.results[0].id]);
+        });
     } else {
-      setValue(defaultValue);
+      setValue(undefined);
     }
-  }
+  }, [props.value.join(',')]);
 
-  const listOptions = () => {
-    return choices
-      .map((item, index) => (
-        <ListOption
-          key={item.id}
-          ref={listOptionRefs[index]}
-          value={item}
-        >
-          {item.label}
-        </ListOption>
-      ));
-  };
-
-  useImperativeHandle(selfRef, () => ({
+  useImperativeHandle(ref, () => ({
+    ...autocompleterRef.current,
     get dirty() {
-      const providedValue = props.value[0] !== defaultValue[0] ? props.value[0] : undefined;
```

Comment:
did this work?

Reply:
It used to work because we had a "reserved default value" of `[-1]` to indicate no choice is selected. We now only need to keep the array empty to indicate that no choice is selected.